src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -93,13 +97,20 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 	public void push(RowMap r) throws Exception {
 		String key = r.pkToJson(keyFormat);
 		String value = r.toJSON();
+		addTableTopic(context.getConfig().useTableTopic, r);
 
 		ProducerRecord<String, String> record =
 				new ProducerRecord<>(topic, this.partitioner.kafkaPartition(r, this.numPartitions), key, value);
 
 		kafka.send(record, new KafkaCallback(r, this.context, key, value));
 	}
 
+	private void addTableTopic(boolean useTableTopic, RowMap r) {
+		if (useTableTopic) {
+			this.topic = this.originalTopic + "-" + r.getDatabase() + "." + r.getTable();
```

Comment:
you don't need to set the instance variable here; it's messy.  better to do something like:

```
   String value = r.toJSON();
   String messageTopic; 
   if ( useTableTopic ) { 
      messageTopic = getTableTopic(...)
  } else {
      messageTopic = this.topic
  }
```